src/raster-api.c

```diff
@@ -0,0 +1,133 @@
+/*!
+ * \file raster-api.c
+ * \date 2024-12-13
+ * \authors Alessandro Bridi [ale.bridi15@gmail.com]
+ *
+ * \brief LibRaster API functions implementations
+ *
+ * \details A rasterizer is a software renderer that works by writing pixels to
+ *      a framebuffer.
+ *      This implementation lets the user define 3 callbacks that defines the
+ *      technique used to write them, so that hardware accelerators can be used
+ *      to archive big speedups.
+ */
+
+#include "raster-api.h"
+#include "fontutils-api.h"
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <inttypes.h>
+
+#define MAX_BUFFER_SIZE (128)
+
+/*!
+ * \brief Draws a text box with background, value, and label
+ *
+ * \details This function draws a text box on the screen using the provided
+ *     drawing callbacks. It handles background color, value formatting,
+ *     threshold-based coloring, slider rendering, and label drawing.
+ * 
+ * \param[in] box Pointer to the Box structure containing the text box configuration
+ * \param[in] draw_rectangle Callback function to draw rectangles
+ * \param[in] line_callback Callback function to draw lines of text
+ */
+void prv_draw_text_box(struct RasterBox *box, raster_draw_rectangle_callback draw_rectangle, font_draw_line_callback line_callback) {
+    if (RASTER_PARTIAL == 0 && !box->updated)
+        return;
+    // Draw the basic rectangle
+    draw_rectangle(box->rect.x, box->rect.y, box->rect.w, box->rect.h, box->color);
+    if (box->label) {
```

Comment:
Always check against `NULL`, a pointer is not a boolean expression.
```suggestion
    if (box->label != NULL) {
```